tests/unittests/GraphOptzTest.cpp

```diff
@@ -331,15 +331,21 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvWithReshapeConst) {
       F_->createBatchNormalization(bindings_, "batch", CV, 3, 0.0001, 0.9);
   F_->createSave("ret", BN);
 
+  EXPECT_EQ(F_->getNodes().size(), 4);
+  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchNormalizationNodeKind), 1);
+
+  optimizedF_ = F_->clone(F_->getName().str() + "_optimized");
   // Initialize to ensure that constant tensors are not optimized out.
   bindings_.allocate(filter)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
   bindings_.allocate(bias)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
-  ::glow::convertPlaceholdersToConstants(F_, bindings_, {input});
+  ::glow::convertPlaceholdersToConstants(optimizedF_, bindings_, {input});
+  ::glow::optimize(optimizedF_, CompilationMode::Infer);
+  EXPECT_EQ(optimizedF_->getNodes().size(), 2);
+  EXPECT_EQ(
+      countNodeKind(optimizedF_, Kinded::Kind::BatchNormalizationNodeKind), 0);
 
-  EXPECT_EQ(F_->getNodes().size(), 4);
-  ::glow::optimize(F_, CompilationMode::Infer);
-  EXPECT_EQ(F_->getNodes().size(), 2);
-  EXPECT_EQ(countNodeKind(F_, Kinded::Kind::BatchNormalizationNodeKind), 0);
+  bindings_.allocate(input)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
+  checkNumericalEquivalence();
```

Comment:
I think there should be a way to use the `optimizeFunction` helper instead of cloning the function and optimizing it explicitly:

```
  EXPECT_EQ(F_->getNodes().size(), 4);
  EXPECT_EQ(
      countNodeKind(F, Kinded::Kind::BatchNormalizationNodeKind), 1);

  bindings_.allocate(filter)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
  bindings_.allocate(bias)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
  ::glow::convertPlaceholdersToConstants(F_, bindings_, {input});

  optimizedF_ = optimizeFunction(F_);
  EXPECT_EQ(optimizedF_->getNodes().size(), 2);
  EXPECT_EQ(
      countNodeKind(optimizedF_, Kinded::Kind::BatchNormalizationNodeKind), 0);

  bindings_.allocate(input)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
  checkNumericalEquivalence();
```

Reply:
@SplitInfinity I'm glad you mentioned this. I tried used the `optimizeFunction` on my first attempt, but saw that the `BatchNormalizationNode` count was still at `1`. More precisely, the code above produces the errors:
```
/Users/padentomasello/code/glow/tests/unittests/GraphOptzTest.cpp:343: Failure                                
Expected equality of these values:                                                                            
  optimizedF_->getNodes().size()                                                                              
    Which is: 3                                                                                               
  2                                                                                                           
/Users/padentomasello/code/glow/tests/unittests/GraphOptzTest.cpp:345: Failure                                
Expected equality of these values:                                                                            
  countNodeKind(optimizedF_, Kinded::Kind::BatchNormalizationNodeKind)
    Which is: 1                                                                                               
  0   
```
I thought the problem was that we needed to call `::glow::convertPlaceholdersToConstants` on `optimizedF_`, before we call `::glow::optimize`, which can't be done using the helper. Does this make sense?